decentdao/fractal-interface
# Implement markdown support for proposal templates and proposal view

Closes #1399 

## Changes:
- Changes to `ProposalTemplates` to display markdown from description. See `/daos/[daoAddress]/proposal-templates/new` page
- Rendering proposals in the list with stripped markdown styling. See `/daos/[daoAddress]/proposals`(Proposals list page) and `/daos/[daoAddress]` (DAO Dashboard - Activity list rendering affected)
- Rendering Markdown in proposal details. See `/daos/[daoAddress]/proposals/[proposalId]`

We certainly will need to revisit this once we adopt `ProposalTemplates` components. Probably we want to implement description preview / editor tab, so instead of visualizing description preview in the sidebar - we should be doing that via switching markdown input "mode" from edit to preview (or somehow displaying both of them).

I also tried to incorporate https://www.npmjs.com/package/@uiw/react-markdown-editor and https://github.com/OpusCapita/react-markdown - yet, I found them quite opinionated and thought it would be better to iterate over editor experience, so pure `TextArea` is the best option as of now, from my opinion

<img width="1593" alt="Знімок екрана 2024-03-11 о 23 36 47" src="https://github.com/decent-dao/fractal-interface/assets/25801162/fd32cc8c-9cbd-498c-8c76-97da989cfc89">
<img width="852" alt="Знімок екрана 2024-03-14 о 13 08 37" src="https://github.com/decent-dao/fractal-interface/assets/25801162/6818bd60-28a0-4d72-aeb7-27b3c53e0745">


## Testing:
1. Create Azorius or Multisig DAO
2. Create proposal with markdown in description. Make sure to put some very long strings to test edge cases
3. Make sure it properly appears in the proposals list, yet it should be truncated. In the list it should be as plain text without markdown styling applied, but also markdown symbols should be removed
4. Make sure markdown properly displayed on proposal details page
